sources/packages/backend/libs/test-utils/src/factories/application.ts

```diff
@@ -48,7 +48,7 @@ export function createFakeApplication(
   application.data = options?.initialValue.data ?? ({} as ApplicationData);
   application.programYear = relations?.programYear ?? createFakeProgramYear();
   // TODO get programYear from relations instead of setting the id here.
-  application.programYear.id = 2;
+  application.programYear.id = options?.initialValue?.programYearId ?? 2;
```

Comment:
Please add a `programYear: programYear` in the relations object and use it from there.
Also, all properties decorated with `@RelationId` will be removed.